ekotrace-debug-collector/src/cli.rs

```diff
@@ -0,0 +1,549 @@
+                session_id: 0.into(),
+                big_endian: false,
+                attach_target: Some("stm32".to_string()),
+                gdb_addr: None,
+                interval: Duration::from_millis(1000),
+                output_path: "./out".into(),
+                tracer_addrs: vec![0x1, 0x10, 0x100, 0x20000000, 0x20000004, 0x20000008]
+            }
+        )
+    }
+
+    /// Should error if both big and little endian specified
+    #[test]
+    fn error_specify_both_endianness() {
+        if let Ok(_) = config_from_options(options_from_str(
+            "ekotrace-debug-collector \
+            --session-id 0 \
+            --little-endian \
+            --big-endian \
+            --attach stm32 \
+            --interval 1s \
+            --output-file ./out \
+            0x100",
+        )) {
+            panic!("No error when specified both big and little endian")
+        }
+    }
+
+    /// Default to little endian if none specified and no ELF given
+    #[test]
+    fn specify_neither_endianness() {
+        compile_symbol_example();
+        assert_eq!(
+            config_from_options(options_from_str(
+                "ekotrace-debug-collector \
+                --session-id 0 \
+                --attach stm32 \
+                --interval 1s \
+                --output-file ./out \
+                0x1"
+            ))
+            .unwrap(),
+            Config {
+                session_id: 0.into(),
+                big_endian: false,
+                attach_target: Some("stm32".to_string()),
+                gdb_addr: None,
+                interval: Duration::from_millis(1000),
+                output_path: "./out".into(),
+                tracer_addrs: vec![0x1]
+            }
+        )
+    }
+
+    /// Imply endianness from ELF if not specified
+    #[test]
+    fn imply_endianness() {
+        compile_symbol_example();
+        assert_eq!(
+            config_from_options(options_from_str(
+                "ekotrace-debug-collector \
+                --session-id 0 \
+                --attach stm32 \
+                --interval 1s \
+                --output-file ./out \
+                --elf \
+                ./tests/symbols-example/target/thumbv7em-none-eabihf/debug/symbols-example \
+                0x1"
+            ))
+            .unwrap(),
+            Config {
+                session_id: 0.into(),
+                big_endian: false,
+                attach_target: Some("stm32".to_string()),
+                gdb_addr: None,
+                interval: Duration::from_millis(1000),
+                output_path: "./out".into(),
+                tracer_addrs: vec![0x1]
+            }
+        );
+        assert_eq!(
+            config_from_options(options_from_str(
+                "ekotrace-debug-collector \
+                --session-id 0 \
+                --attach stm32 \
+                --interval 1s \
+                --output-file ./out \
+                --elf ./tests/example-be-elf \
```

Comment:
I switched to a runtime-compiled crate for testing the symbol parsing, but I think it makes sense for endianness detection to just include a small binary. It's just 600 bytes and basically just consists of the ELF header that is read to figure out endianness, so there's not any reason we'd want to change this.